src/components/views/settings/SetIdServer.tsx

```diff
@@ -393,28 +374,27 @@ export default class SetIdServer extends React.Component<IProps, IState> {
 
         return (
             <SettingsFieldset legend={sectionTitle} description={bodyText}>
-                <form className="mx_SetIdServer" onSubmit={this.checkIdServer}>
-                    <Field
+                <TooltipProvider>
```

Comment:
I decided that rather than trying to work around Field's validiity, we should just update this setting to use `EditInPlace`. This means we're using a Compound component, with support for asynchronous actions and built in support for progress and error text.

Ultimately this feels like a much better change.